File: pinyougou_parent/pinyougou_sellergoods_interface/src/main/java/com/pinyougou/sellergoods/service/ContentCategoryService.java
package com.pinyougou.sellergoods.service;
import java.util.List;
import com.pinyougou.pojo.TbContentCategory;							  
import entity.PageResult;
import entity.Result;
/**
 * contentCategory服务层接口
 * @author jieweili
 *
 */
public interface ContentCategoryService {

    /**
     * 返回全部列表(已弃置)
     * @return
     */
    List<TbContentCategory> findAll();
 
    /**
     * 返回分页对象(过时)
     * @param pageNum
     * @param pageSize
     * @return
     */
	 
    PageResult findPage(Integer pageNum, Integer pageSize);

    /**
     * 添加数据
     * @param contentCategory
     * @return
     */
    public Result saveContentCategory(TbContentCategory contentCategory);

    /**
     * 根据id查询一个对象
     * @param id
     * @return
     */
    TbContentCategory getContentCategoryById(Long id);

    /**
     * 修改对象
     * @param contentCategory
     * @return
     */
    Result updateContentCategory(TbContentCategory contentCategory);

    /**
     * 批量删除
     * @param idList
     * @return
     */
    Result removeContentCategory(Long[] idList);

    /**
     * 根据条件分页查询
     * @param contentCategory
     * @param pageNum
     * @param pageSize
     * @return
     */
    PageResult listContentCategoryByExample(TbContentCategory contentCategory, Integer pageNum, Integer pageSize);
 
}

File: pinyougou_parent/pinyougou_manager_web/src/main/webapp/js/service/contentCategoryService.js
app.service('contentCategoryService', function ($http) {
    //旧的查询方法(弃置)								   
    this.findPage = function (pageNum, pageSize) {
        return $http.get('../contentCategory/findPage.do?pageNum=' + pageNum + '&pageSize=' + pageSize)
    };

    //查询方法,读取列表数据绑定到表单中
    this.search = function (pageNum, pageSize, searchEntity) {
        return $http.post('../contentCategory/findPage.do?pageNum=' + pageNum + "&pageSize=" + pageSize, searchEntity)
    };

    //删除方法
    this.removeContentCategoryList = function (idList) {
        return $http.get('../contentCategory/removeContentCategory.do?idList=' + idList)
    };

    //根据id查询
    this.findOne = function (id) {
        return $http.get('../contentCategory/findOne.do?id=' + id)
    };

    //修改品牌
    this.updateContentCategory = function (contentCategory) {
        return $http.post('../contentCategory/updateContentCategory.do', contentCategory)
    };

    //新增品牌
    this.saveContentCategory = function (contentCategory) {
        return $http.post('../contentCategory/saveContentCategory.do', contentCategory)
    };   
});
